### DISCUSSION TITLE
Append the extension to assets' id

### DISCUSSION THREAD

**Discussion by nagman** [directus/directus#3510] [Feature Requests]
With the API, we can get every asset providing its id: `/assets/4f8g4df1gdf518-5sdf4g8sdf5-dsf4f8g4d`

But in some cases, we need the extension to be at the end of the url, for instance when working with `<video>`.
If we provide an `src` without an extension - and even if we explicitly provide the mime type - the `video` element is unable to control the `currentTime` position when user interacts.

I made a jsfiddle, with the first video being requested from the directus api (without extension) and the second one being a simple video of big buck bunny (with extension): https://jsfiddle.net/8vnub75e/1/

And a small video to demonstrate what I mean:
https://user-images.githubusercontent.com/16116486/103086498-165db780-45e5-11eb-9ce9-67273dacab16.mp4

This is not due to my video, it's only because there's no extension in the `src` (I've made the same test with an absolute url to the video file and everything works).

**Comment by nagman** [comment]
Actually the video issue isn't related to extensions. I've made a separate post for that: https://github.com/directus/directus/issues/3640

But having the extension in the URL is still needed for some cases, like if you use the [`react-svg`](https://www.npmjs.com/package/react-svg) package to get svg files. It will throw an error because it can't find the `.svg` extension.

Here's the solution I've found to get around this issue:
```
# Add this in your apache config file, if you run directus behind a reverse proxy

RewriteEngine On
RewriteRule (.*)\.(jpg|gif|png|mp3|mp4) $1 [PT]
```

This will recognize an url like `/assets/887g8d7fg8df-df6gd4f6-fg6sdf4g6.mp4` an proxy the request to `/assets/887g8d7fg8df-df6gd4f6-fg6sdf4g6`.
So that you can add extensions to your assets URLs (it doesn't matter if the extension is not the correct one).

It's not a solution, 'cause you may or may not run directus behind a reverse proxy, and you could use any extension even if it's not the correct one.

**Reply by philleepflorence** [reply]
See related discussion for solution - [Allow file extensions on asset URL's #4036](https://github.com/directus/directus/discussions/4036)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
